### PR TITLE
Add TikTak global optimization to sciml_train

### PR DESCRIPTION
**Pull request by ranjanan**

### CODE REVIEW

File: src/train.jl [hunk 1]
```diff
@@ -355,3 +355,51 @@ function sciml_train(loss, opt::BBO = BBO(), data = DEFAULT_DATA;lower_bounds, u
                                         NaN,
                                         bboptre.elapsed_time)
 end
+
+function sciml_train(loss, opt::TikTak, data = DEFAULT_DATA;lower_bounds, upper_bounds, local_method,
```
Comment (ChrisRackauckas) [inline src/train.jl:359]: initial condition for uniformity?

File: src/train.jl [hunk 2]
```diff
@@ -355,3 +355,51 @@ function sciml_train(loss, opt::BBO = BBO(), data = DEFAULT_DATA;lower_bounds, u
                                         NaN,
                                         bboptre.elapsed_time)
 end
+
+function sciml_train(loss, opt::TikTak, data = DEFAULT_DATA;lower_bounds, upper_bounds, local_method,
+                      maxiters = get_maxiters(data), kwargs...)
```
Comment (ChrisRackauckas) [inline src/train.jl:360]: any way to do callback support?

File: src/train.jl (hunk 1 above)
Comment (ranjanan) [reply src/train.jl:359]: Sure

File: src/train.jl (hunk 2 above)
Comment (ranjanan) [reply src/train.jl:360]: No, there does not seem to be any callback support.

**Comment by ChrisRackauckas** [comment]
test?

**Comment by ChrisRackauckas** [comment]
test it in the neural network context like the others: https://github.com/SciML/DiffEqFlux.jl/blob/master/test/layers_sciml.jl#L91-L144 . It would probably be good to refactor that whole deal to its own file though.

**Comment by ChrisRackauckas** [comment]
seems like overkill for 3.

**Pull request merged**